report-viewer/src/model/factories/ComparisonFactory.ts

```diff
@@ -3,16 +3,57 @@ import type { Match } from '../Match'
 import type { SubmissionFile } from '../SubmissionFile'
 import type { MatchInSingleFile } from '../MatchInSingleFile'
 import store from '@/stores/store'
+import { generateColors } from '@/utils/ColorUtils'
+import slash from 'slash'
 
 /**
  * Factory class for creating Comparison objects
  */
 export class ComparisonFactory {
+  public static getComparison(id1: string, id2: string) {
+    console.log('Generating comparison {%s} - {%s}...', id1, id2)
+    let comparison = new Comparison('', '', 0)
+
+    //getting the comparison file based on the used mode (zip, local, single)
```

Comment:
the complexity / nesting depth of the following code is quite high. I would propose splitting it up. This is especially true regarding the HTTP request that should always be isolated